pkg/controller/drain/drain_controller.go

```diff
@@ -381,9 +456,34 @@ func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) erro
 			ctrl.enqueueAfter(node, ctrl.cfg.DrainRequeueDelay)
 		}
 
+		nErr := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateDrained), Message: fmt.Sprintf("Node Drain has failed")},
+			&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateDrained, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateExecuted), string(v1alpha1.MachineConfigNodeUpdateDrained)), Message: fmt.Sprintf("Node Drain has failed. Error is: %s The drain will not be complete until desired drainer %s matches current drainer %s", err.Error(), node.Annotations[constants.DesiredDrainerAnnotationKey], node.Annotations[constants.LastAppliedDrainerAnnotationKey])},
+			metav1.ConditionUnknown,
+			metav1.ConditionUnknown,
+			node,
+			ctrl.client,
+			ctrl.featureGatesAccessor,
+		)
+		if nErr != nil {
+			klog.Errorf("Error making MCN: %w", nErr)
```

Comment:
I know this isn't finalized, but a suggestion I would have to make debugging easier is to make unique error messages in the logs